src/GuppyDataExplorer/ExplorerTable/index.jsx

```diff
@@ -224,6 +226,45 @@ class ExplorerTable extends React.Component {
       });
     });
   };
+  /**
+   * Sets empty coumns visability to state showEmptyColumns
+   * @param colList: column configs for root table
+   * @returns column configs for root table with
+   * empty coumns visability set to state showEmptyColumns
+   */
+  markEmptyColums = (colList) => {
+    if (!this.props.rawData || this.props.rawData.length === 0) {
+      return colList;
+    }
+    // Go through each column and see if it has any current data
+    return colList.map((obj) => {
+      const tempObj = obj;
+      let columnIsEmpty = true;
+      // see if any item has data in current column
+      for (let i = 0; i < this.props.rawData.length; i += 1) {
```

Comment:
minor point, you might be able to simplify this loop code with using some helper functions from `lodash` like `findIndex` https://lodash.com/docs/4.17.15#findIndex

Reply:
simplified using every